daffodil-lib/src/main/scala/org/apache/daffodil/util/Memoize1.scala

```diff
@@ -60,4 +61,11 @@ class Memoize1[-T <: AnyRef, +R] private (f: T => R) extends (T => R) {
 
 object Memoize1 {
   def apply[T <: AnyRef, R](f: T => R) = new Memoize1(f).apply _
+
+  private var count: Long = 0L
+
+  def size = count
+  def reset {
+    count = 0
+  }
```

Comment:
Doesn't even look like we use Memoize except for in tests making sure it works. It's an interesting class, but we don't seem to need it? Thoughts on getting rid of it? It seems to have decent tests, so it's not a big deal if we keep it.

Reply:
While we are not using it, I did use it extensively while figuring this stuff out, and eventually removed it in favor of the SharedFactory idiom which is more purpose built, and so easier to understand. 

I will remove it from daffodil-lib however, and just try to keep in in some side-pony project for future use.